pkg/controller/maps/controller_test.go

```diff
@@ -91,6 +98,11 @@ func TestReconcileMapsServer_Reconcile(t *testing.T) {
 			post: func(r ReconcileMapsServer) {
 				// watches should have been cleared
 				require.Empty(t, r.DynamicWatches().Secrets.Registrations())
+
+				// observedGeneration should not have been updated
+				var ems v1alpha1.ElasticMapsServer
+				require.NoError(t, r.Get(context.Background(), types.NamespacedName{Name: nsnFixture.Name, Namespace: nsnFixture.Namespace}, &ems))
+				require.Equal(t, int64(1), ems.Status.ObservedGeneration)
```

Comment:
Given that this is repeated many times, you could have a little helper function defined in the scope of the test that you just call here.

At the top:
```golang
func TestReconcileMapsServer_Reconcile(t *testing.T) {
	timeFixture := metav1.Now()

	assertObservedGeneration := func(r k8s.Client, expected int) {
		var ems v1alpha1.ElasticMapsServer
		require.NoError(t, r.Get(context.Background(), types.NamespacedName{Name: nsnFixture.Name, Namespace: nsnFixture.Namespace}, &ems))
		require.Equal(t, int64(expected), ems.Status.ObservedGeneration)
	}
```
and then here
```golang
			post: func(r ReconcileMapsServer) {
				// watches should have been cleared
				require.Empty(t, r.DynamicWatches().Secrets.Registrations())
				// observedGeneration should not have been updated
				assertObservedGeneration(r, 1)
			},
```